### PR TITLE
Use Github Artifacts v4

### PR DESCRIPTION
**Pull request by t3chguy**
---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: .github/workflows/docs-pr-netlify.yaml [hunk 1]
```diff
@@ -11,13 +11,10 @@ jobs:
         if: github.event.workflow_run.conclusion == 'success' && github.event.workflow_run.event == 'pull_request'
         runs-on: ubuntu-latest
         steps:
-            # There's a 'download artifact' action, but it hasn't been updated for the workflow_run action
-            # (https://github.com/actions/download-artifact/issues/60) so instead we get this mess:
             - name: 📥 Download artifact
-              uses: dawidd6/action-download-artifact@e7466d1a7587ed14867642c2ca74b5bcc1e19a2d # v3
+              uses: actions/download-artifact@v4
```
Comment (dbkr) [inline .github/workflows/docs-pr-netlify.yaml:15]: Shouldn't we still be referencing the sha so it we get a well defined version?

Comment (t3chguy) [reply .github/workflows/docs-pr-netlify.yaml:15]: We compromise for highly trusted actions between security & convenience, given that Github can always serve us whatever they like however we specify the ref.

Comment (dbkr) [reply .github/workflows/docs-pr-netlify.yaml:15]: Ah okay, fair enough.

**Review by dbkr — COMMENTED** [review]
Tests seem unhappy though: is there a path with a literal asterisk in it that it's trying to upload or has it got confused?

**Review by dbkr — APPROVED** [review]

**Pull request merged**